src/main/java/List.java

```diff
@@ -0,0 +1,148 @@
+import com.sun.source.util.TaskListener;
+
+public class List {
```

Comment:
I would suggest not using `List` as the name since collections have an interface called `List`. It can potentially trip the user of your codebase.
Secondly, List doesn't specify what kind of list is this. 
Suggested: Name it to reflect the purpose